logicrunner/logicrunner.go

```diff
@@ -413,7 +417,38 @@ func (lr *LogicRunner) ProcessExecutionQueue(ctx context.Context, es *ExecutionS
 		}
 
 		finish()
+
+		if lr.pulseChanged(ctx, es, *qe.parcel.Message().DefaultTarget()) {
+			return
+		}
+	} // for
+}
+
+// pulseChanged checks whether last execution was a pending one.
+// If this is true as a side effect the function sends a PendingFinished
+// message to the current executor
+func (lr *LogicRunner) pulseChanged(ctx context.Context, es *ExecutionState, currentRef core.RecordRef) bool {
+	es.Lock()
+	defer es.Unlock()
+
+	if !es.pending {
+		return false
+	}
+
+	es.pending = false
+	msg := message.PendingFinished{Reference: currentRef}
```

Comment:
I don't understand how this classifies as "pulseChanged".

Reply:
OK, fixed